src/easylink/pipeline_schema_constants/main.py

```diff
@@ -0,0 +1,334 @@
+"""
+=============================
+Main EasyLink Pipeline Schema
+=============================
+"""
+
+from easylink.graph_components import (
+    EdgeParams,
+    InputSlot,
+    InputSlotMapping,
+    OutputSlot,
+    OutputSlotMapping,
+)
+from easylink.step import (
+    EmbarrassinglyParallelStep,
+    HierarchicalStep,
+    InputStep,
+    LoopStep,
+    OutputStep,
+    ParallelStep,
+    Step,
+)
+from easylink.utilities.aggregator_utils import concatenate_datasets
+from easylink.utilities.splitter_utils import split_data_by_size
+from easylink.utilities.validation_utils import (
+    dont_validate,
+    validate_clusters,
+    validate_ids_to_remove,
+    validate_input_dataset,
+    validate_datasets_directory,
+    validate_links,
+)
+
+NODES = [
+    InputStep(output_slots=[OutputSlot("input_datasets"), OutputSlot("known_clusters")]),
+    LoopStep(
+        template_step=HierarchicalStep(
```

Comment:
whoa, so you're looping a single mega-step for all of "entity_resolution"?

Reply:
We decided to call this "entity resolution" instead of "clustering pass" to make it more familiar to people who _don't_ do cascading! That's also why I decided to call the overall pipeline schema "main" (to disambiguate it from this step).